### PR TITLE
welsh and new link

### PR DESCRIPTION
**Pull request by sealDebolina**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PRL-6037
https://tools.hmcts.net/jira/browse/PRL-6094
https://tools.hmcts.net/jira/browse/PRL-6108
https://tools.hmcts.net/jira/browse/PRL-6109
https://tools.hmcts.net/jira/browse/PRL-6110

### Change description ###
PRL-6037:CUI - Display doc links in welsh on the dashboard
PRL-6094:CUI - Amends to displaying links on the DA Applicant dashboard tasklist
PRL-6108:CUI - Amends to displaying links on the CA Applicant dashboard tasklist
PRL-6109:CUI - Amends to displaying links on the CA Respondent dashboard tasklist
PRL-6110:CUI - Amends to displaying links on the DA Applicant dashboard tasklist

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/steps/common/task-list/components/tasklist/utils.ts [hunk 1]
```diff
@@ -145,6 +153,11 @@ export const getYourWitnessStatementStatus = (userCase: Partial<CaseWithId>): St
 export const getCheckAllegationOfHarmStatus = (caseData): StateTags => {
```
Comment (vivek-sekhar) [inline src/main/steps/common/task-list/components/tasklist/utils.ts:153]: pass lang param and use the same method for both

File: src/main/steps/common/task-list/components/tasklist/utils.ts [hunk 2]
```diff
@@ -145,6 +153,11 @@ export const getYourWitnessStatementStatus = (userCase: Partial<CaseWithId>): St
...
   return _.get(caseData, 'c1ADocument.document_binary_url') ? StateTags.READY_TO_VIEW : StateTags.NOT_AVAILABLE_YET;
 };
+export const getCheckAllegationOfHarmStatusWelsh = (caseData): StateTags => {
```
Comment (vivek-sekhar) [inline src/main/steps/common/task-list/components/tasklist/utils.ts:156]: get rid of it

File: src/main/steps/common/task-list/components/tasklist/utils.ts [hunk 3]
```diff
@@ -195,3 +208,13 @@ export const getInternationalFactorsStatus = (
 export const getFinalApplicationStatus = (caseData): StateTags => {
   return _.get(caseData, 'finalDocument.document_binary_url') ? StateTags.READY_TO_VIEW : StateTags.NOT_AVAILABLE_YET;
```
Comment (vivek-sekhar) [inline src/main/steps/common/task-list/components/tasklist/utils.ts:209]: use same method for both lang

File: src/main/steps/common/task-list/components/tasklist/utils.ts [hunk 4]
```diff
@@ -195,3 +208,13 @@ export const getInternationalFactorsStatus = (
...
 };
+export const getFinalApplicationWelshStatus = (caseData): StateTags => {
+  return _.get(caseData, 'finalWelshDocument.document_binary_url')
+    ? StateTags.READY_TO_VIEW
+    : StateTags.NOT_AVAILABLE_YET;
+};
```
Comment (vivek-sekhar) [inline src/main/steps/common/task-list/components/tasklist/utils.ts:215]: get rid of this

File: src/main/steps/common/task-list/components/tasklist/utils.ts [hunk 5]
```diff
@@ -195,3 +208,13 @@ export const getInternationalFactorsStatus = (
...
+export const isRespondentSubmitedResponse = (caseData: Partial<CaseWithId>) => {
```
Comment (vivek-sekhar) [inline src/main/steps/common/task-list/components/tasklist/utils.ts:216]: Not required

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/common/documents/download/DownloadDocumentController.ts [hunk 6]
```diff
@@ -14,69 +14,85 @@ export default class DownloadDocumentController {
...
-            doc.partyId === userDetails.id &&
-            doc.categoryId === DocumentCategory.RESPONDENT_C7_RESPONSE_TO_APPLICATION &&
-            doc.documentLanguage === 'cy'
-        )?.document;
+        if (language === 'en') {
+          documentReference = caseData?.respondentDocuments?.find(
+            doc =>
+              doc.partyId === userDetails.id &&
+              doc.categoryId === DocumentCategory.RESPONDENT_C7_RESPONSE_TO_APPLICATION &&
+              doc.documentLanguage === 'en'
```
Comment (vivek-sekhar) [inline src/main/steps/common/documents/download/DownloadDocumentController.ts:52]: check against lang. no need of if else

File: src/main/steps/common/documents/download/DownloadDocumentController.ts [hunk 7]
```diff
@@ -14,69 +14,85 @@ export default class DownloadDocumentController {
...
-        documentReference = caseData?.finalDocument;
+        if (language === 'en') {
+          documentReference = caseData?.finalDocument;
+        } else {
+          documentReference = caseData.finalWelshDocument;
+        }
         break;
       case 'aoh-document':
-        documentReference = caseData?.c1ADocument;
+        if (language === 'en') {
```
Comment (vivek-sekhar) [inline src/main/steps/common/documents/download/DownloadDocumentController.ts:40]: use ternary

File: src/main/steps/common/documents/download/DownloadDocumentController.ts [hunk 8]
```diff
@@ -14,69 +14,85 @@ export default class DownloadDocumentController {
...
+          documentReference = caseData?.finalDocument ?? caseData.c100DraftDoc;
+        } else {
+          documentReference = caseData.finalWelshDocument ?? caseData.c100DraftDocWelsh;
+        }
         break;
 
       case 'fl401-application':
       case 'cada-document':
-        documentReference = caseData?.finalDocument;
+        if (language === 'en') {
```
Comment (vivek-sekhar) [inline src/main/steps/common/documents/download/DownloadDocumentController.ts:33]: use ternary

File: src/main/steps/common/documents/download/DownloadDocumentController.ts [hunk 9]
```diff
@@ -14,69 +14,85 @@ export default class DownloadDocumentController {
...
-    userDetails: UserDetails
+    userDetails: UserDetails,
+    language: string
   ): { documentId: string; documentName: string } {
     let documentReference;
 
     switch (documentType) {
       case 'c100-application':
-        documentReference = caseData?.finalDocument ?? caseData.c100DraftDoc;
+        if (language === 'en') {
```
Comment (vivek-sekhar) [inline src/main/steps/common/documents/download/DownloadDocumentController.ts:24]: use ternary

File: src/main/steps/common/documents/download/DownloadDocumentController.ts [hunk 10]
```diff
@@ -14,69 +14,85 @@ export default class DownloadDocumentController {
...
-        )?.document;
-        break;
-      case 'c1a-application-document-welsh':
-        documentReference = caseData?.respondentDocuments?.find(
-          doc =>
-            doc.partyId === userDetails.id &&
-            doc.categoryId === DocumentCategory.RESPONDENT_C1A_RESPONSE_TO_APPLICATION &&
-            doc.documentLanguage === 'cy'
-        )?.document;
+        if (language === 'en') {
```
Comment (vivek-sekhar) [inline src/main/steps/common/documents/download/DownloadDocumentController.ts:64]: check against lang. no need of if else

File: src/main/steps/common/documents/download/DownloadDocumentController.ts [hunk 11]
```diff
@@ -14,69 +14,85 @@ export default class DownloadDocumentController {
...
-        break;
-      case 'c1a-response-document-welsh':
-        documentReference = caseData?.respondentDocuments?.find(
-          doc =>
-            doc.partyId === userDetails.id &&
-            doc.categoryId === DocumentCategory.RESPONDENT_RESPOND_TO_C1A &&
-            doc.documentLanguage === 'cy'
-        )?.document;
+        if (language === 'en') {
+          documentReference = caseData?.respondentDocuments?.find(
```
Comment (vivek-sekhar) [inline src/main/steps/common/documents/download/DownloadDocumentController.ts:82]: check against lang. no need of if else

File: src/main/steps/common/task-list/components/tasklist/config/ca_applicant.ts [hunk 12]
```diff
@@ -91,12 +92,58 @@ export const CA_APPLICANT: TaskListConfigProps[] = [
...
+        openInAnotherTab: () => true,
+      },
+      {
+        //** validate **
+        id: Tasks.YOUR_AOH_PDF_WELSH,
+        href: () =>
+          applyParms(DOWNLOAD_DOCUMENT_BY_TYPE, {
+            partyType: PartyType.APPLICANT,
+            documentType: 'aoh-document',
+            language: 'cy',
```
Comment (vivek-sekhar) [inline src/main/steps/common/task-list/components/tasklist/config/ca_applicant.ts:140]: use contant for lang

File: src/main/steps/common/task-list/components/tasklist/utils.ts [hunk 13]
```diff
@@ -192,6 +202,13 @@ export const getInternationalFactorsStatus = (
   return StateTags.TO_DO;
 };
 
-export const getFinalApplicationStatus = (caseData): StateTags => {
-  return _.get(caseData, 'finalDocument.document_binary_url') ? StateTags.READY_TO_VIEW : StateTags.NOT_AVAILABLE_YET;
+export const getFinalApplicationStatus = (caseData, lang): StateTags => {
+  return _.get(caseData, lang === 'en' ? 'finalDocument.document_binary_url' : 'finalWelshDocument.document_binary_url')
```
Comment (vivek-sekhar) [inline src/main/steps/common/task-list/components/tasklist/utils.ts:206]: use constant for lang

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/common/documents/download/DownloadDocumentController.ts (hunk 7 above)
Comment (sealDebolina) [reply src/main/steps/common/documents/download/DownloadDocumentController.ts:40]: nested ternary oparator will create sonar issue

File: src/main/steps/common/documents/download/DownloadDocumentController.ts (hunk 8 above)
Comment (sealDebolina) [reply src/main/steps/common/documents/download/DownloadDocumentController.ts:33]: nested ternary oparator will create sonar issue

File: src/main/steps/common/documents/download/DownloadDocumentController.ts (hunk 9 above)
Comment (sealDebolina) [reply src/main/steps/common/documents/download/DownloadDocumentController.ts:24]: nested ternary oparator will create sonar issue

**Review by vivek-sekhar — APPROVED** [review]

**Pull request merged**